ckanext/dcatapchharvest/dcat_helpers.py

```diff
@@ -22,3 +26,66 @@ def uri_to_iri(uri):
 def get_langs():
     language_priorities = ['en', 'de', 'fr', 'it']
     return language_priorities
+
+
+def dataset_uri(dataset_dict):
+    """
+    Returns an URI for the dataset
+
+    This will be used to uniquely reference the dataset on the RDF
+    serializations.
+
+    The value will be the first found of:
+
+        1. The value of the `uri` field
+        2. The value of an extra with key `uri`
+        3. 'http://' . [org part of identifier] . '/' . [first part of identifier]
+            where dataset identifier looks like '123@org-name'
```

Comment:
I think the problem with your idea is that a uri should have two properties: 
- it should be unique
- the dataset should get served under that address, or at least there must be a redirect to where the dataset is served from. 

With your uri 3. I don't see how that dataset will be served on that uri: 
I think you can take instead the ckan-url of the dataset: ckan-site-url/dataset/dataset['id'] to arrive at a uri. WDYT?